### PR TITLE
chore: add balances shared library for retrieving and updating balances

### PR DESCRIPTION
**Pull request by guoguojin**

### CODE REVIEW

File: pools/pools.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+	"github.com/govalues/decimal"
+	redisv9 "github.com/redis/go-redis/v9"
+)
+
+// Pool represents a liquidity pool in the DORA network.
+// This struct is for serialization purposes only.
+type Pool struct {
+	BaseAsset     string          `json:"base_asset" redis:"base_asset"`
+	QuoteAsset    string          `json:"quote_asset" redis:"quote_asset"`
+	IsProductPool bool            `json:"is_product_pool" redis:"is_product_pool"`
```
Comment (kosegor) [inline pools/pools.go:22]: Not sure if we need this one

**Review by kosegor — COMMENTED** [review]
Let's see if we can figure out the optimal Balance and Balances structures

File: pools/pools.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	AmountShares  uint64          `json:"amount_shares" redis:"amount_shares"`
+	AmountBase    uint64          `json:"amount_base" redis:"amount_base"`
+	AmountQuote   uint64          `json:"amount_quote" redis:"amount_quote"`
```
[the remark this answers is not in the document]
Comment (guoguojin) [reply pools/pools.go:37]: As previously stated, this struct matches the data expected for the match.Pool type so I'm going to leave the naming as is for now, it's just so we can serialize and deserialize the data to/from Redis, the match.Pool type only has private fields and accessors making it unsuitable for serialization, hence this struct.

File: pools/pools.go [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+	FeeFactor     decimal.Decimal `json:"fee_factor" redis:"fee_factor"`
```
[the remark this answers is not in the document]
Comment (guoguojin) [reply pools/pools.go:38]: According to the match.Pool struct in the bond-api:

```
// feeFactor functions as SwapFee for product pool, and as G for yieldspace pool
feeFactor decimal.Decimal
```

**Review by kosegor — APPROVED** [review]

**Pull request merged**